project/shih-chuan/README.md

```diff
@@ -0,0 +1,95 @@
+# Raycast2D
+
+## Basic Information
+
+Simulate how lights radiate from a single light source on a 2D map.
+
+* github repository: [https://github.com/shih-chuan/Raycast2D](https://github.com/shih-chuan/Raycast2D)
+
+## Problem to Solve
+
+The problem this system wants to solve is to calculate all the areas where lights from a given single light source can reach on a 2D map.
+
+There are some algorithms can be implemented for solving this problem. Before we dive into the topic deeper, let's first talk about the Line of sight(LOS) algorithm, which tells us how to simulate the travel of a "single" ray casted from a light source.
+
+* Line of sight(LOS) algorithm: Input the x,y coordinates for the target and the light source and then walk along a line between them. At each coordinates on the line, the algorithm checks if anything in the map blocks the light from reaching the target.
+
+However, a light source doesn't only cast a single ray, but it radiates infinite rays. To solve the problem, a naive approach is to cast rays to every potential destination, but this is time consuming. We can use the following algorithms, provided by the [redblobgames](https://www.redblobgames.com/articles/visibility/), to improve the efficiency.
+
+* Ray casting: Cast the rays only at angles where the walls begin or end, and fills the triangles produced by these rays.
+* Wall Tracking: The above algorithm has a problem, it has to intersect with every single wall. Combining the ray casting and wall intersection into a single algorithm is a more efficient approach. This can be achieved by sweeping around 360° and keeping track of the walls that intersect the sweep line. Here is the pseudo code:
+
+    ```pseudo
```

Comment:
I like the way to use pseudo code to describe the problem.  Since you plan to use Python in the high-level, it makes sense to replace the pseudo code using Python too.  It may later become unit tests.

Reply:
OK, I've replaced the pseudo code with python.

Reply:
I hope you feel to have a clearer idea about what you want to do  :)